packages/runtime/src/controls/image.ts

```diff
@@ -1,6 +1,23 @@
+import { P, match } from 'ts-pattern'
 import { CopyContext, ReplacementContext } from '../state/react-page'
 
-export type ImageControlData = string
+type ImageControlDataV0 = string
+
+type MakeswiftFileImageData = {
+  type: 'makeswift-file'
+  id: string
+}
+
+type ExternalFileImageData = {
+  type: 'external-file'
+  url: string
+  width?: number | null
+  height?: number | null
+}
```

Comment:
Didn't we have versions in these at some point?

Reply:
We did for the Image prop controller. I can add them in here, but they won't be used as of now.

Reply:
Well, the point would be so that when we have future versions that also use an object we can discriminate between those and this ones. It's not essential but it would be nice for us to have that... eventually, even having all the types in there with the type of the control might allow us to do introspection without even having the schema in the backend 🤔